app/models/site_configuration.rb

```diff
@@ -17,29 +17,21 @@ def self.instance
 
   # This override allows discarding of reasons and updating them by reason_code
   # if a discarded reason of a particular code is found and updated, it will be undiscarded.
-  # TODO: move into separate service?
   def revision_reasons_attributes=(attributes)
     attributes.each do |attribute, _|
-      if attribute["_discard"] == true
-        self.revision_reasons.find(attribute["id"]).discard if attribute["id"].present?
+      if attribute["id"].present?
+        # Update the existing record
+        existing_record = self.revision_reasons.find(attribute["id"])
+        existing_record.update(attribute.except("id"))
       else
-        if attribute["id"].present?
-          # Update the existing record
-          existing_record = self.revision_reasons.with_discarded.find(attribute["id"])
-          existing_record.undiscard if existing_record.discarded?
-          existing_record.update(attribute.except("id"))
-        else
-          reason_code = attribute["reason_code"]
-          discarded_reason = self.revision_reasons.with_discarded.find_by(reason_code: reason_code)
+        existing_reason = self.revision_reasons.with_discarded.find_by(reason_code: attribute["reason_code"])
 
-          if discarded_reason&.discarded?
-            # Undiscard and update the record
-            discarded_reason.undiscard
-            discarded_reason.update(attribute.except("id"))
-          else
-            # Proceed with the normal behavior
-            self.revision_reasons.build(attribute)
-          end
+        if existing_reason.present?
+          # Undiscard and update the record
+          existing_reason.update(attribute.except("id").merge({ discarded_at: nil }))
+        else
+          # Proceed with the normal behavior
+          self.revision_reasons.build(attribute)
         end
```

Comment:
replace this with and see if it works. If not original way is fine but this way the update is done by rails active record
```
 def revision_reasons_attributes=(attributes)
    attributes.each do |attribute, _|
      next unless attribute["id"].blank? && attribute["reason_code"].present?

      existing_reason = self.revision_reasons.with_discarded.find_by(reason_code: attribute["reason_code"])

      next unless existing_reason.present?

      # Undiscard and update the record
      attribute["id"] = existing_reason.id
      attribute["discarded_at"] = nil
    end

    super(attributes)
  end
```

Reply:
remember when using nested attributes, have to call save on the parent model